File: generate-icons.js
const fs = require('fs');
const path = require('path');

// Create icons directory if it doesn't exist
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir);
}

// SVG template for a camera/screenshot icon
function createIconSVG(size) {
  return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#667eea;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#764ba2;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Background circle -->
  <circle cx="${size/2}" cy="${size/2}" r="${size/2}" fill="url(#grad)"/>
  
  <!-- Camera body -->
  <rect x="${size*0.25}" y="${size*0.35}" width="${size*0.5}" height="${size*0.4}" 
        rx="${size*0.05}" fill="white" opacity="0.9"/>
  
  <!-- Lens -->
  <circle cx="${size/2}" cy="${size*0.55}" r="${size*0.15}" 
          fill="white" opacity="0.6"/>
  
  <!-- Flash/viewfinder -->
  <rect x="${size*0.65}" y="${size*0.3}" width="${size*0.08}" height="${size*0.08}" 
        rx="${size*0.02}" fill="white" opacity="0.8"/>
</svg>`;
}

// Icon sizes to generate
const sizes = [16, 32, 48, 128];

sizes.forEach(size => {
  const svg = createIconSVG(size);
  const filename = path.join(iconsDir, `icon${size}.svg`);
  fs.writeFileSync(filename, svg);
  console.log(`✓ Generated icon${size}.svg`);
});

console.log('\n📸 SVG icons generated successfully!');
console.log('\nTo convert to PNG (you have options):');
console.log('\n1. Online converter:');
console.log('   - Visit https://svgtopng.com');
console.log('   - Upload each SVG file');
console.log('   - Download as PNG with same filename');
console.log('\n2. Using ImageMagick (if installed):');
console.log('   convert icons/icon16.svg icons/icon16.png');
console.log('   convert icons/icon32.svg icons/icon32.png');
console.log('   convert icons/icon48.svg icons/icon48.png');
console.log('   convert icons/icon128.svg icons/icon128.png');
console.log('\n3. Using Inkscape (if installed):');
console.log('   inkscape icons/icon16.svg -o icons/icon16.png');
console.log('   inkscape icons/icon32.svg -o icons/icon32.png');
console.log('   inkscape icons/icon48.svg -o icons/icon48.png');
console.log('   inkscape icons/icon128.svg -o icons/icon128.png');
console.log('\nAlternatively, you can use any image editor to create simple PNG icons.');

File: generate-icons.py
#!/usr/bin/env python3
"""
Generate placeholder PNG icons for the Chrome extension.
Requires: pip install pillow
"""

import os
from PIL import Image, ImageDraw

def create_icon(size, filename):
    """Create a simple camera icon."""
    # Create image with gradient background
    img = Image.new('RGB', (size, size), color='#667eea')
    draw = ImageDraw.Draw(img)
    
    # Calculate proportional dimensions
    center = size // 2
    camera_width = int(size * 0.5)
    camera_height = int(size * 0.4)
    camera_x = (size - camera_width) // 2
    camera_y = int(size * 0.35)
    
    lens_radius = int(size * 0.15)
    flash_size = int(size * 0.08)
    
    # Draw camera body (white rectangle)
    draw.rounded_rectangle(
        [camera_x, camera_y, camera_x + camera_width, camera_y + camera_height],
        radius=size // 20,
        fill='white'
    )
    
    # Draw lens (circle)
    lens_y = int(size * 0.55)
    draw.ellipse(
        [center - lens_radius, lens_y - lens_radius, 
         center + lens_radius, lens_y + lens_radius],
        fill='#764ba2'
    )
    
    # Draw flash/viewfinder (small rectangle)
    flash_x = int(size * 0.65)
    flash_y = int(size * 0.3)
    draw.rounded_rectangle(
        [flash_x, flash_y, flash_x + flash_size, flash_y + flash_size],
        radius=size // 40,
        fill='#ffeb3b'
    )
    
    # Save the image
    img.save(filename, 'PNG')
    print(f'✓ Created {filename}')

def main():
    # Create icons directory
    icons_dir = 'icons'
    os.makedirs(icons_dir, exist_ok=True)
    
    # Generate all required sizes
    sizes = [16, 32, 48, 128]
    
    print('📸 Generating PNG icons...\n')
    
    for size in sizes:
        filename = os.path.join(icons_dir, f'icon{size}.png')
        create_icon(size, filename)
    
    print('\n✨ All icons generated successfully!')
    print(f'Icons saved to: {os.path.abspath(icons_dir)}/')

if __name__ == '__main__':
    try:
        main()
    except ImportError:
        print('❌ Error: Pillow library not found')
        print('Install it with: pip install pillow')
        print('\nOr create icons manually using any image editor:')
        print('  - Create 4 PNG images: 16x16, 32x32, 48x48, 128x128')
        print('  - Simple camera/screenshot icon')
        print('  - Save in icons/ folder')
